heartbeat/tests/system/test_monitor.py

```diff
@@ -41,6 +41,43 @@ def test_http(self, status_code):
             raise SkipTest
         self.assert_fields_are_documented(output[0])
 
+    @parameterized.expand([
+        ("up", '{"foo": {"baz": "bar"}}'),
+        ("down", '{"foo": "unexpected"}'),
+        ("down", 'notjson'),
+    ])
+    def test_http_json(self, expected_status, body):
+        """
+        Test JSON response checks
+        """
+        server = self.start_server(body, 200)
+        try:
+            self.render_config_template(
+                monitors=[{
+                    "type": "http",
+                    "urls": ["http://localhost:{}".format(server.server_port)],
+                    "check_response_json": [{
+                        "description": "foo equals bar",
+                        "condition": {
+                            "equals": {"foo": {"baz": "bar"}}
```

Comment:
What does the YAML output look like here? I assume if you use `"foo.baz": "bar"` things will not work? What if the response actually has this exact key with a dot inside?

Sorry  to keep poking on this one. Agree it's a potential condition issue if it breaks but it shows up here very prominent.

Reply:
I did some investigation and determined that dotted paths and nested maps are equivalent here due to the use of `common.MapStr`. This means that you can use either in your condition and it will work.

I think we should move further discussion of this to a separate issue since this has more to do with the beats condition evaluation engine than this specific PR.